internal/k8s/client.go

```diff
@@ -235,6 +229,13 @@ func maybeImmutableFieldStderr(stderr string) bool {
 // Currently ignores any "not found" errors, because that seems like the correct
 // behavior for our use cases.
 func (k K8sClient) Delete(ctx context.Context, entities []K8sEntity) error {
+	l := logger.Get(ctx)
+	var b strings.Builder
+	for _, e := range entities {
+		b.WriteString(fmt.Sprintf("Deleting via kubectl: %s/%s\n", e.Kind.Kind, e.Name()))
+	}
+	l.Infof(b.String())
```

Comment:
This will crash if the string contains any % signs. 

You should either use Infof inside the loop (rather than have a separate Printf) or use something like Infof("%s", string)